string name
string id_name
---
uint32 nid
